assets/js/blocks/cart-checkout/checkout/checkout-error/index.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { CART_URL } from '@woocommerce/block-settings';
+import { Icon, exclamation } from '@woocommerce/icons';
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Gets the checkout error from the server if possible.
+ */
+const CheckoutErrorMessage = () => {
+	const checkoutData = getSetting( 'checkoutData', {} );
+
+	if ( typeof checkoutData.message === 'string' ) {
+		return (
+			<p className="wc-block-checkout-error__description">
+				{ checkoutData.message }
+			</p>
+		);
+	}
+
+	return (
+		<p className="wc-block-checkout-error__description">
+			{ __(
+				'There was a problem creating an order from your cart items. Please go back to your cart and fix any reported problems before retrying.',
+				'woo-gutenberg-products-block'
+			) }
+		</p>
+	);
+};
+
+/**
+ * Renders checkout error state.
+ */
+const CheckoutError = () => {
+	return (
+		<div className="wc-block-checkout-error">
+			<Icon
+				className="wc-block-checkout-error__image"
+				alt=""
+				srcElement={ exclamation }
+				size={ 100 }
+			/>
+			<strong className="wc-block-checkout-error_title">
+				{ __(
+					'There was a problem with your cart',
+					'woo-gutenberg-products-block'
+				) }
+			</strong>
+			<CheckoutErrorMessage />
+			<span className="wp-block-button">
+				<a href={ CART_URL } className="wp-block-button__link">
+					{ __( 'View your cart', 'woo-gutenberg-products-block' ) }
+				</a>
```

Comment:
The `CheckoutError` component appears to be setup to handle _any_ error that prevents checkout from rendering. Given that, will a `View Cart` link always be relevant to the error?  Will the "...problem with your cart" message always be relevant?

It _appears_ that this is tailored to error's having cart related issues as the root cause. With that in mind, I wonder if I'm getting tripped up more by the component name and a different name might work better? Something like `CheckoutCartError` or `CartError`?

Reply:
Ill adjust this and add some more generic button/message handling based on error code.